pkg/credentialprovider/plugin/examples/ecr/main.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// TODO: this is just a reference implementation, this plugin should be moved
+// to k8s.io/cloud-provider-aws
+package main
+
+import (
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"net/url"
+	"os"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/credentialprovider/apis/criauth/v1alpha1"
+	plugin "k8s.io/kubernetes/pkg/credentialprovider/plugin/framework"
+)
+
+var ecrPattern = regexp.MustCompile(`^(\d{12})\.dkr\.ecr(\-fips)?\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?$`)
+
+type ecrPlugin struct {
+}
+
+func (e *ecrPlugin) GetAuthPluginResponse(image string, extraArgs map[string]string) (*v1alpha1.CRIAuthPluginResponse, error) {
```

Comment:
Won't block on this since its easy to do in a follow-up, but in order to debug, there should be a way to send the output of the entire response from ECR to stderr, i.e. look for a verbosity flag on extraArgs.